### ISSUE TITLE
copysign function inside @avx loop

### ISSUE THREAD

**Issue by henry2004y** [chriselrod/LoopVectorization.jl#105]
Hi,

Fantastic package!
When I recently tried to use some sign functions inside the loop, for example
```
using LoopVectorization, BenchmarkTools

function mydotavx(a, b)
    s = 0.0
    @avx for i ∈ eachindex(a,b)
        s += a[i]*b[i]
        s += minmod(a[i],b[i])
    end
    s
end

minmod(a, b) = (sign(a) + sign(b)) / 2.0 * min(abs(a), abs(b))

a = rand(256); b = rand(256);

@btime mydotavx($a, $b)
```
there comes the error message:
```
ERROR: LoadError: MethodError: no method matching copysign(::NTuple{4,VecElement{Float64}}, ::NTuple{4,VecElement{Float64}})
Stacktrace:
 [1] macro expansion at /Users/hyzhou/.julia/packages/SIMDPirates/Mppjk/src/floating_point_arithmetic.jl:27 [inlined]
 [2] sign at /Users/hyzhou/.julia/packages/SIMDPirates/Mppjk/src/SIMDPirates.jl:80 [inlined]
 [3] minmod(::VectorizationBase.SVec{4,Float64}, ::VectorizationBase.SVec{4,Float64}) at /Users/hyzhou/Documents/Computer/Julia/Dante/test_loop.jl:12
 [4] macro expansion at /Users/hyzhou/.julia/packages/LoopVectorization/bI2Fm/src/reconstruct_loopset.jl:444 [inlined]
 [5] _avx_! at /Users/hyzhou/.julia/packages/LoopVectorization/bI2Fm/src/reconstruct_loopset.jl:444 [inlined]
 [6] mydotavx(::Array{Float64,1}, ::Array{Float64,1}) at /Users/hyzhou/Documents/Computer/Julia/Dante/test_loop.jl:5
...
```

I guess what is missing here is that we need to implement a new copysign function?
I am also wondering what we should do for more functions inside the loop like this.
Thanks!

**Comment by chriselrod** [comment]
Hi, thanks for the bug report.
Turns out my old `sign` function was broken.

Early on in SIMDPirate's history, shortly after it was forked from SIMD, it tried to do everything with `NTuple{W,Core.VecElement{T}}`s instead of wrapping them in structs. The primary motivation was that on some platforms, wrapping them in structs would often cause crashes when they crossed function boundaries. That bug has since been fixed, so now there's little reason not to wrap them in structs. `@benchmark`/`@btime` on anything returning a struct-wrapped `NTuple{W,Core.VecElement{T}}` was very likely to crash Julia (on those platforms).

But because ``NTuple{W,Core.VecElement{T}}` is a type defined in `Base` Julia, I can't overload methods defined in base Julia without commiting type piracy, so every function was given a name like `vcopysign`.
But that meant I couldn't overload any base methods, like `copysign`, as that'd be type piracy, so `vcopysign` is defined on `NTuple{W,Core.VecElement{T}}` instead, while `copysign` is defined on `SVec`s (which is a struct wrapping such tuples).

tldr: `copysign` was already defined, but there was a bug preventing it from being called.

I'll close this issue when I make it depend on the new SIMDPirates version with the fix.
```julia
julia> using LoopVectorization, BenchmarkTools
[ Info: Precompiling LoopVectorization [bdcacae8-1622-11e9-2a5c-532679323890]

julia> @inline minmod(a, b) = (sign(a) + sign(b)) / 2.0 * min(abs(a), abs(b)) # @inline seems necessary
minmod (generic function with 1 method)

julia> function mydot(a, b)
           s = 0.0
           @inbounds @simd for i ∈ eachindex(a,b)
               s += a[i]*b[i]
               s += minmod(a[i],b[i])
           end
           s
       end
mydot (generic function with 1 method)

julia> function mydotavx(a, b)
           s = 0.0
           @avx for i ∈ eachindex(a,b)
               s += a[i]*b[i]
               s += minmod(a[i],b[i])
           end
           s
       end
mydotavx (generic function with 1 method)

julia> a = rand(256); b = rand(256);

julia> @btime mydotavx($a, $b)
  54.581 ns (0 allocations: 0 bytes)
151.08749163626982

julia> @btime mydot($a, $b)
  92.835 ns (0 allocations: 0 bytes)
151.08749163626982

julia> a = rand(255); b = rand(255);

julia> @btime mydotavx($a, $b)
  54.419 ns (0 allocations: 0 bytes)
139.69162315609353

julia> @btime mydot($a, $b)
  159.942 ns (0 allocations: 0 bytes)
139.6916231560935
```

**Comment by chriselrod** [comment]
Bumped SIMDPirates version on master, and registering.
